_posts/2020-07-22-#15644.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: pr
+date: 2020-07-22
+title: "Make orphan processing interruptible"
+pr: 15644
+authors: [sipa]
+components: ["p2p"]
+host: narula
+status: upcoming
+commit:
+---
+
+
+## Notes
+
+* [PR 15644](https://github.com/bitcoin/bitcoin/pull/15644) was merged
+  over a year ago. It changes the way a node looks through its orphan
+  set to figure out if orphans can now be processed and added to the
+  mempool.
+
+* A transaction is an _orphan_ if we have not yet seen any of its
+  inputs. This might be a valid transaction, but we don't know yet. We
+  have to be careful with orphan management because orphans cannot be
+  validated.
+
+* Orphans are stored in a map called `mapOrphanTransactions` by their
+  txid. We limit the number of orphan transactions to [`-maxorphantx`]
+  (https://github.com/sipa/bitcoin/blob/9453018fdc8f02d42832374bcf1d6e3a1df02281/src/net_processing.h#L17)
+  which is by default 100.
+
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+   You're always encouraged to put your PR review on GitHub, even after it has
+   been merged.
+
+2. Why might we want a node to keep track of orphans at all? What does
+   this help with?
+
+3. Look at the way orphans are added to the orphan maps and how
+   orphans are re-evaluated when we accept new transactions to the
+   mempool. What happens when a node receives a transaction where it
+   has not seen the transaction's inputs?
+
+4. What is stored in `mapOrphanTransactionsByPrev` and why do you
```

Comment:
I've changed this slightly to state the type of `mapOrphanTransactionsByPrev` here since the syntax is quite intimidating.